honeycomb-core/src/cmap2/advanced_ops.rs

```diff
@@ -0,0 +1,87 @@
+//! Advanced operations implementation
+//!
+//! This module contains code used to implement advanced operations, e.g. some non-standard,
+//! higher-level abstractions that are useful in meshing algorithms.
+
+// ------ IMPORTS
+
+use crate::{CMap2, CoordsFloat, DartIdentifier, EdgeIdentifier, Vertex2, NULL_DART_ID};
+
+// ------ CONTENT
+
+impl<T: CoordsFloat> CMap2<T> {
+    /// Split an edge into to segments.
+    ///
+    /// <div class="warning">
+    /// This implementation is 2D specific.
+    /// </div>
+    ///
+    /// This method takes all darts of an edge and rebuild connections in order to create a new
+    /// point on this edge. The position of the point default to the midway point, but it can
+    /// optionally be specified.
+    ///
+    /// In order to minimize editing of embedded data, the original darts are kept to their
+    /// original vertices while the new darts are used to model the new point.
+    ///
+    /// # Panics
+    ///
+    /// This method may panic if the edge upon which the operation is performed does not have two
+    /// defined vertices.a
+    pub fn split_edge(&mut self, edge_id: EdgeIdentifier, midpoint_vertex: Option<Vertex2<T>>) {
+        let d1 = edge_id as DartIdentifier;
+        let d2 = self.beta::<2>(d1);
+        // (*): unwrapping is ok because you probably shouldn't be using this method
+        //      if both vertices of the edge aren't defined
+        if d2 == NULL_DART_ID {
```

Comment:
Not clear: what does it means ?
variable names can be better than d1 and d2.

Reply:
I tried to clarify the comments (e471d51), but I'm not sure about how to name both of these darts